src/MarginModels/MarginBase.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+Created on May 13 2021
+
+@author: mandd,wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import os
+import abc
+import sys
+import numpy as np
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import mathUtils as utils
+from utils import InputData, InputTypes
+#Internal Modules End-
+
+class MarginBase():
```

Comment:
I think it is probably a good time to start to build a common base class for ReliabilityModels, MaintenanceModels, and MarginModels. Currently, in each base of these models, we have implemented almost the same functionalities. Either for this PR or next PR, I think we need to start to use the common base class. Please let me know if you want me to work on it.